EIPS/eip-4844.md

```diff
@@ -64,7 +64,7 @@ Compared to full data sharding, this EIP has a reduced cap on the number of thes
 | Type | Base type | Additional checks |
 | - | - | - |
 | `BLSFieldElement` | `uint256` | `x < BLS_MODULUS` |
-| `Blob` | `Vector[BLSFieldElement, FIELD_ELEMENTS_PER_BLOB]` | |
+| `Blob` | `Vector[BLSFieldElement, FIELD_ELEMENTS_PER_BLOB]` | Each field element encoded as padded 32 bytes big endian |
```

Comment:
the way forward here would be the same as any other BLS data: treat it as a black-boxed array of bytes and specify the serialization elsewhere (rather than bending the ssz interpretation etc)

Reply:
fair point, actually its `uint256` list/array so so talking about big endian encoding is irrelevant.

and with now the rlp serialization, it will anyway serialize it big endian for EL layer

Reply:
well, `BLSFieldElement` in that world needs to be `Bytes32` - `uint256` would imply a SSZ "number" which is little endian.

Reply:
That's indeed how it's defined in consensus-specs: https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#custom-types

This might be one of the desync issues between EIP and consensus-specs.

Am I right that the `Blob` struct is currently unused in the EIP?

Reply:
Blob right now in EIP is only used for network tx format correctness verification via kzg libs

Reply:
> well, `BLSFieldElement` in that world needs to be `Bytes32` - `uint256` would imply a SSZ "number" which is little endian.

Well Uint256 does imply a number which is the intend as its a point on the BLS curve, but i don't think it implies encoding format case in point the below lines from the EIP which were anyway previously the part of EIP.

```
    # Return FIELD_ELEMENTS_PER_BLOB and BLS_MODULUS as padded 32 bytes big endian values
    return Bytes(U256(FIELD_ELEMENTS_PER_BLOB).to_be_bytes32() + U256(BLS_MODULUS).to_be_bytes32())
```

But anyway I think this extra clarification I added in a separate PR takes care of any confusion:
https://github.com/ethereum/EIPs/pull/7038/files#diff-c72ce77807acfea9e9693b7999f247841df938c9645be37013903805fa8b0c09R283

Reply:
> but i don't think it implies encoding format

https://github.com/ethereum/consensus-specs/blob/dev/ssz/simple-serialize.md#uintn

Reply:
Can we use CL format `ByteVector[BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB]` for `Blob` here? `ByteVector` (just bytes) allows you not to go deep into SSZ in EL than `Vector`.